runtime/v2/shim/util_unix.go

```diff
@@ -81,7 +81,9 @@ func AnonDialer(address string, timeout time.Duration) (net.Conn, error) {
 
 // AnonReconnectDialer returns a dialer for an existing socket on reconnection
 func AnonReconnectDialer(address string, timeout time.Duration) (net.Conn, error) {
-	return AnonDialer(address, timeout)
+	ctx, cancel := context.WithTimeout(context.TODO(), timeout)
+	defer cancel()
+	return dialer.ContextDialerFunc(ctx, address, nil)
```

Comment:
why not just use https://golang.org/pkg/net/#DialTimeout ?

Reply:
If we use `DialTimeout` directly, there is no meaning to unify dialer. Also, the dialer implementation is different between *unix and windows.

Reply:
I just found that we can unify the `AnonDialer` and `AnonReconnectDialer` to call `ContextDialer`, then they can be removed from util_unix.go and util_windows.go
Shall I add another commit to unify the Dialer implementation on windwos ?

Reply:
I think it is fine to keep them separated. ContextDialer looks like for grpc call.

Reply:
> I think it is fine to keep them separated. ContextDialer looks like for grpc call.

I mean we can use the `ContextDialer` to unify the `AnonDialer` and `AnonReconnectDialer `'s implementation in `util` on different platforms.

Reply:
Well, I don't think we should align containerd client's dialer with shim's. `DRY` is good principle. But in this case, the `ContextDialer` looks like complicated  than before and the `AnonDialer` should be simple. 

I am not windows developer so I am not sure we can remove the `AnonDialer` from different platforms. Just my two cents.

Reply:
For the shim, there is no necessary to use `ContextDialer` which bring one more goroutines to handle that error. 
The `DialTimeout` is more clear than `ContextDialer` in this case currently. 

Since there are more issues reporting this case, I would like to fix that case and then doing that refactor(unify or something). 

cc @containerd/reviewers @containerd/maintainers

Reply:
I'm also trying to figure out why we drop the given context to call go net.dialer with a timeout value.. which then uses a new background context

We should probably be calling net.DialContext instead but we'd have to go review the history of this to see why that didn't happen..?

Reply:
> I'm also trying to figure out why we drop the given context to call go net.dialer with a timeout value.

The reason I guess is that `DialContext` func has a receiver and it can not be directly used by simply importing `net` package, maybe the maintainer didn't notice it and invented the wheels by himself.
Using `DialContext` could save us lots of effort, I will update this PR later, trying to adapt `DialContext`.